docs/features/collections/collections.md

```diff
@@ -0,0 +1,304 @@
+---
+id: collections
+title: Collections
+sidebar_label: Collections
+---
+
+## What is a collection?
+
+A collection is a data structure with functionality built around it.  Think about a collection as a grouping of similar things.  All the enemies in a game, the items in the inventory, the textures being drawn, images in an animation, or bullets on the screen.
+
+Collections are more advanced than a simple array or list, and each one can provide unique functionality.  There are benefits and disadvantages to each collection type.  Don't use a collection without a reason, especially if a simple array or list will suffice.  Make sure the collection you are choosing has a purpose that is advantageous to you and your project.
+
+.NET has many default Collection types (List, Dictionary, Queue, Stack, and many more).  These are a few additional collections.
+
+## General requirements for all collections
+
+All of these require adding the using directive for `MonoGame.Extended.Collections` in your file.
+```csharp
+using MonoGame.Extended.Collections;
+```
+
+## Bag
+
+A `Bag` is an un-ordered array of items with fast `Add` and `Remove` properties.
+
+### Bag Functionality and Behavior
+- It is much faster than an array when removing items.
+- Takes less space than a linked list.
+- Bag will resize itself (grow by 1.5 times current size) only when it needs to.  
+- Used space won't shrink after removing elements.
+- You can clear the bag.
+- Can't sort the bag.
+- Order is not preserved.
+- Elements are added to the end of the collection.
+
+### Bag Example
+
+The example below creates a new `Bag` of size 3 with type `int`, adds three `integers` to the bag, adds a fourth item, and removes one item. When the fourth item is added this causes a resize of the capacity by 1.5 times the current capacity.
+
+```csharp
+var bag = new Bag<int>(3);
+bag.Add(4);
+bag.Add(8);
+bag.Add(15);
+// bag is now [4, 8, 15]
+
+bag.Add(16); 
+// bag is extended here, capacity is now 4 instead of 3 with items [4, 8, 15, 16]
+
+bag.RemoveAt(1);
+// bag is now [4, 16, 15] with a capacity of 4
```

Comment:
@AristurtleDev  I think that it wouldn't be appropriate unless we were going to dive into the inner workings of the Bag class further.  The reason is that when you access it via the Enumerator, it doesn't let you access that element because of the MoveNext() method inside the BagEnumerator struct.  

The _Items is private and not accessible to an end user.

You can see the inaccessibility of the item by the example sample output which shows it doesn't print anything.

![image](https://github.com/user-attachments/assets/1d91db1d-eeab-483d-970b-f710fcf26ebe)

Also, the default behavior if the element doesn't exist you try to access is to just return the default.

![image](https://github.com/user-attachments/assets/05998d49-e5a4-44d3-b511-62ccce31eda1)

Reply:
Thanks for the clarifications, it would be better to leave as it.